lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb

```diff
@@ -35,6 +39,8 @@ def install!
               create_copy_resources_script if target.includes_resources?
               create_acknowledgements
               create_dummy_source(native_target)
+              remove_build_setting_overrides(config_hash, native_target)
+              create_xcconfig_file(native_target, Xcodeproj::Config.new(config_hash))
```

Comment:
this is moved to the end so each installation step has a chance to modify `config_hash`